src/StackExchange.Redis/RedisDatabase.cs

```diff
@@ -3980,6 +4000,83 @@ protected override RedisValue[] Parse(in RawResult result, out int count)
             }
         }
 
+        private static SortedSetRangeAndStoreCommandMessage CreateSortedSetRangeStoreMessage(
+            int db,
+            CommandFlags flags,
+            RedisKey destinationKey,
+            RedisKey sourceKey,
+            RedisValue start,
+            RedisValue stop,
+            SortedSetOrder sortedSetOrder,
+            Order order,
+            Exclude exclude,
+            long skip,
+            long take)
+        {
+            if (sortedSetOrder == SortedSetOrder.ByRank)
+            {
+                if (take > 0)
+                {
+                    throw new ArgumentException("take argument is not valid when sortedSetOrder is ByRank you may want to try setting the SortedSetOrder to ByLex or ByScore", nameof(take));
+                }
+
+                if (exclude != Exclude.None)
+                {
+                    throw new ArgumentException("exclude argument is not valid when sortedSetOrder is ByRank, you may want to try setting the sortedSetOrder to ByLex or ByScore", nameof(exclude));
+                }
+            }
+
+            var arguments = new List<RedisValue>();
+            arguments.Add(exclude == Exclude.Both || exclude == Exclude.Start ? $"({start}" : sortedSetOrder == SortedSetOrder.ByLex ? $"[{start}" : start);
+            arguments.Add(exclude == Exclude.Both || exclude == Exclude.Stop ? $"({stop}" : sortedSetOrder == SortedSetOrder.ByLex ?  $"[{stop}" : stop);
+
+            if (sortedSetOrder != SortedSetOrder.ByRank)
+            {
+                arguments.Add(sortedSetOrder.GetLiteral());
```

Comment:
Instead, let's call `.GetLiteral()` and check for `.IsNull` on our `if` here - extra line, but cleaner and decoupled.

Reply:
Sure thing, that's how I initially had it, I figured an enum-comparison would be slightly more efficient than the null check.